components/esp_modem/src/esp_modem_command_library.cpp

```diff
@@ -560,4 +562,280 @@ command_result set_gnss_power_mode_sim76xx(CommandableIf *t, int mode)
     return generic_command_common(t, "AT+CGPS=" + std::to_string(mode) + "\r");
 }
 
+command_result get_gnss_information_sim70xx(CommandableIf *t, gps_t& gps)
+{
+    ESP_LOGV(TAG, "%s", __func__ );
+    std::string_view out;
+    auto ret = generic_get_string(t, "AT+CGNSINF\r", out);
+    if (ret != command_result::OK) {
+        return ret;
+    }
+
+
+
+    constexpr std::string_view pattern = "+CGNSINF: ";
+    if (out.find(pattern) == std::string_view::npos) {
+        return command_result::FAIL;
+    }
+    /**
+     * Parsing +CGNSINF:
+     * <GNSS run status>,
+     * <Fix status>,
+     * <UTC date &  Time>,
+     * <Latitude>,
+     * <Longitude>,
+     * <MSL Altitude>,
+     * <Speed Over Ground>,
+     * <Course Over Ground>,
+     * <Fix Mode>,
+     * <Reserved1>,
+     * <HDOP>,
+     * <PDOP>,
+     * <VDOP>,
+     * <Reserved2>,
+     * <GNSS Satellites in View>,
+     * <Reserved3>,
+     * <HPA>,
+     * <VPA>
+     */
+    out = out.substr(pattern.size());
+    int pos = 0;
+    if ((pos = out.find(',')) == std::string::npos) {
+    	return command_result::FAIL;
+    }
+    //GNSS run status
+    int GNSS_run_status;
+    if (std::from_chars(out.data(), out.data() + pos, GNSS_run_status).ec == std::errc::invalid_argument) {
+        return command_result::FAIL;
+    }
+    gps.run = (gps_run_t)GNSS_run_status;
+    out = out.substr(pos + 1);
+    if ((pos = out.find(',')) == std::string::npos) {
+    	return command_result::FAIL;
+    }
+    //Fix status
+    {
+		std::string_view fix_status = out.substr(0 , pos);
```

Comment:
This change/file needs a cleanup in general, at least for formatting and mixed spaces+tabs